### DISCUSSION TITLE
What is the correct place to put Google Adsense code?

### DISCUSSION THREAD

**Discussion by ArvindRSingh** [razonyang/hugo-theme-bootstrap#673] [Q&A] [answered]
The Google AdSense suggests adding code in the `<HEAD>` but HBS adds it at the end of the `<BODY>`.
Maybe Google has changed the place to add code for new enrollments.

After adding googleAdsense in params.yaml the AdSense coded is added at the end as part of the body.

```
<html>
<body>
<!--StartFragment-->

<script data-ad-client=ca-pub-345345345 async src=https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js></script><script>'serviceWorker'in navigator&&window.addEventListener('load',()=>{navigator.serviceWorker.register('https://semicolon.today/service-worker.min.js').then(function(a){console.log('Successfully registered service worker',a)}).catch(function(a){console.warn('Error whilst registering service worker',a)})})</script>
--
  | </body>
  | </html>

<!--EndFragment-->
</body>
</html>
```

This is what Google suggests
![image](https://user-images.githubusercontent.com/2056238/185884630-569060a6-223b-40f8-b238-39d421f7d1e7.png)

### Concern
As Google AdSense expects code in `<HEAD>` it might reject the enrollment.

Should this be changed in the HBS code, or is my finding wrong?

**Comment by razonyang** [comment]
Let's follow the suggestion of Google AdSense, it shouldn't a problem, since the script has `async` attribute.

See #674.

**Reply by razonyang** [reply, marked as answer]
@ArvindRSingh  BTW, the Google AdSense script seems doesn't work well with service worker cache(unable to load), I suggest upgrading theme to latest commit which won't cache third-party scripts.
